Source/CarthageKit/Project.swift

```diff
@@ -728,6 +754,37 @@ private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL:
 		|> take(1)
 }
 
+/// Sends the URL to each bcsymbolmap found in the given directory.
+private func BCSymbolMapsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL)
+		|> filter { URL in URL.pathExtension == "bcsymbolmap" }
+}
+
+/// Sends the URLs of the bcsymbolmap files that match the given framework and are
+/// located somewhere within the given directory.
+private func BCSymbolMapsForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return UUIDsForFramework(frameworkURL)
+		|> flatMap(.Merge) { UUIDs in
+			if UUIDs.isEmpty {
+				return SignalProducer.empty
+			}
+			return BCSymbolMapsInDirectory(directoryURL)
+				|> { (signal: Signal<NSURL, CarthageError>) -> Signal<NSURL, CarthageError> in
```

Comment:
This seems a bit hacky, returning a new SignalProducer would be better to understand?

```swift
return UUIDsForFramework(frameworkURL)
  |> flatMap(.Merge) { UUIDs in
    if UUIDs.isEmpty {
      return .empty
    }
  
    return SignalProducer { observer, disposable in
      var remainingUUIDs = UUIDs
      BCSymbolMapsInDirectory(directoryURL)
        |> filter { fileURL in
          if let fileUUID = fileURL.lastPathComponent?.stringByDeletingPathExtension.flatMap { NSUUID(UUIDString: $0) } {
            return remainingUUIDs.remove(fileUUID) != nil
          } else {
            return false
          }
        }
        |> take(UUIDs.count)
        |> startWithSignal { signal, signalDisposable in
          disposable += signalDisposable
          signal.observe(observer)
        }
    }
  }
```